### PR TITLE
[BACKPORT] Documenting aggregations. Adding tests for null corner cases.

### PR DESCRIPTION
**Pull request by tombujok**
Backport of https://github.com/hazelcast/hazelcast/pull/9901

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/aggregation/CountAggregationTest.java [hunk 1]
```diff
@@ -71,4 +71,40 @@ public void testCountAggregator_withAttributePath() {
...
+        List<Person> values = samplePersons();
+        values.add(null);
+        long expectation = values.size();
+
+        Aggregator<Map.Entry<Person, Person>, Long> aggregation = Aggregators.count("age");
+        for (Person person : values) {
+            aggregation.accumulate(createEntryWithValue(person));
+        }
+
+        Aggregator<Map.Entry<BigDecimal, BigDecimal>, Long> resultAggregation = Aggregators.count();Aggregators.count("age");
```
Comment (vbekiaris) [inline hazelcast/src/test/java/com/hazelcast/aggregation/CountAggregationTest.java]: `Aggregation.count();` is redundant here and shades the actual assignment to `Aggregation.count("age");`.
Same issue exists in existing code in `CountAggregationTest:68`

Comment (tombujok) [inline hazelcast/src/test/java/com/hazelcast/aggregation/CountAggregationTest.java]: Thx, fixed.
